libgit2-sys/lib.rs

```diff
@@ -391,7 +402,7 @@ pub struct git_fetch_options {
     pub version: c_int,
     pub callbacks: git_remote_callbacks,
     pub prune: git_fetch_prune_t,
-    pub update_fetchhead: c_int,
+    pub update_flags: c_uint,
```

Comment:
You're right, I wasn't familiar with C bitfields, I originally thought they were just a convenient way of specifying bit-flags which are handled in a similar way to bitmasks for unsigned ints. It does seem hard to support since their memory layout is compiler dependent. I was thinking since we know the number of bits set, assuming that bits are packed only either left-to-right or right-to-left, we could do something like `update_flag.bits() | update_flag.bits().reverse_bits()` when getting the unsigned int as a workaround? (This admittedly doesn't work if the memory layout is different.)

Reply:
@ehuss whats the way to interact with c bitfields via FFI?

Reply:
There is no official support for C bitfields in Rust: https://github.com/rust-lang/rfcs/issues/314.

Reply:
Sorry, I haven't had much time lately. I've been intending to push a change that will use a C wrapper to get/set the fields, I just haven't gotten to it, yet. I don't feel comfortable trying to guess the layout from within Rust. @bnjmnt4n If you have time to give that a shot, that would be helpful. Otherwise, I'll try to look into it soonish.

Reply:
@ehuss @bnjmnt4n The maintainer of libgit2 said he can guarantee the ABI will use least significant bits for the flags: https://github.com/libgit2/libgit2/issues/6800

Reply:
Status update: ethomson mentioned in https://github.com/libgit2/libgit2/issues/6800 that they will look to see if there is some workaround for us.

Reply:
I will update this PR when I'm free, since https://github.com/libgit2/libgit2/pull/6806 was merged, but I'm guessing we'll still want to wait for a new release of `libgit2`?